### ISSUE TITLE
Reviewing some TODOs

### ISSUE THREAD

**Issue by AhmetCanSolak** [CSBDeep/CSBDeep#27]
Can somebody with more experience on project review these lines for a possible cleanup?

https://github.com/CSBDeep/CSBDeep/blob/29daf01c0a4e870f191838262ad913b67b9e18dd/csbdeep/data/generate.py#L71-L83

**Comment by uschmidt83** [comment]
Hi, thanks for taking an interest! What do you want to do?

**Comment by uschmidt83** [comment]
Hi @AhmetCanSolak,

are you trying to improve the internal code quality? You're more than welcome to address the few TODOs in the code and suggest helpful refactoring. However, end users might only benefit slightly from that.

What is your motivation to improve the code and how much time can or do you want to spend on this?
For example, adding/updating tests would be useful. Same goes for improving the documentation.

Best,
Uwe

**Comment by AhmetCanSolak** [comment]
> are you trying to improve the internal code quality? You're more than welcome to address the few TODOs in the code and suggest helpful refactoring. However, end users might only benefit slightly from that.
> 

Yes you are right, for most of the users benefit will not be much. But it is always good to keep things consistent and less confusing, especially for possible users who want to play around the code.

> What is your motivation to improve the code and how much time can or do you want to spend on this?
> For example, adding/updating tests would be useful. Same goes for improving the documentation.
> 

I love to have a look on tests as well. Right now i am only considering to have a look in my free time. Main motivation is just adding a new couple eyes to repo and improve internal codebase where it is possible.

**Comment by uschmidt83** [comment]
> Yes you are right, for most of the users benefit will not be much. But it is always good to keep things consistent and less confusing, especially for possible users who want to play around the code.

I generally agree. However, this is currently not high on my list of priorities.

> I love to have a look on tests as well. Right now i am only considering to have a look in my free time. Main motivation is just adding new couple eyes to repo and improve internal codebase where it is possible.

I'm open to suggestions.

Best,
Uwe
